mavisp/modules.py

```diff
@@ -1819,3 +1821,95 @@ def ingest(self, mutations):
             raise MAVISpMultipleError(warning=warnings,
                                       critical=[])
 
+class EnsembleAllosigmaPSN(MavispModule):
+
+    module_dir = "long_range_psn"
+    name = "long_range_psn"
+    method_dir = "allosigma2"
+    exp_files = {'allosigma_mut.txt': {'df_name': 'simple_data',
+                    'columns': ['wt_residue', 'position', 'mutated_residue', 'allosigma-mode']},
+            'results_summary.txt': {'df_name': 'ensemble_data',
+                    'columns': ['Variant_Site', 'Total_Paths']}}
+
+    def _generate_allosigma_psn_classification(self, row):
+        res_num = row['res_num']
+        allosigma_mode = row['allosigma-mode']
+        variant_sites = self.ensemble_data[self.ensemble_data['Variant_Site'].astype(str) == res_num]
+        
+       # For mutations not predicted to have effects
+        if variant_sites.empty: 
+            return 'neutral' if allosigma_mode in ['UP', 'DOWN'] else 'uncertain'
+                
+        # For mutations with predicted effects
+        total_paths = variant_sites['Total_Paths'].max()
+        if total_paths >= 1:
+            return 'damaging' if allosigma_mode in ['UP', 'DOWN'] else 'neutral'
+        else:
+            return 'uncertain'
+
+    def _read_file(self, file_path, df_name, columns):
+        try:
+            # Read the file with the specified columns
+            df = pd.read_csv(file_path, sep='\t', usecols=columns)
+        except Exception as e:
+            this_error = f"Error reading file {file_path}: {e}"
```

Comment:
`parsing` here would be better than `reading` - it is more general, reading implies problems with reading the file from disk (for instance because of permission issues or file system error etc)